src/tools/clippy/clippy_utils/src/ty.rs

```diff
@@ -214,36 +214,21 @@ pub fn implements_trait<'tcx>(
     trait_id: DefId,
     args: &[GenericArg<'tcx>],
 ) -> bool {
-    let callee_id = cx
-        .enclosing_body
-        .map(|body| cx.tcx.hir().body_owner(body).owner.to_def_id());
```

Comment:
Using `body_owner.owner` is almost never the right thing to do. Would using `body_owner_def_id` solve the issue directly?

Reply:
Please forgive me but is that question directed towards me? Please note that the code you've cited is deleted by this PR.